server/src/internalClusterTest/java/org/elasticsearch/index/engine/MaxDocsLimitIT.java

```diff
@@ -61,6 +63,18 @@ protected Collection<Class<? extends Plugin>> nodePlugins() {
         return CollectionUtils.appendToCopy(super.nodePlugins(), TestEnginePlugin.class);
     }
 
+    @Override
+    protected Settings nodeSettings(int nodeOrdinal, Settings otherSettings) {
+        // Document page size should not be too small, else we can fail to write the cluster state for small max doc values
+        return Settings.builder()
+            .put(super.nodeSettings(nodeOrdinal, otherSettings))
+            .put(
+                PersistedClusterStateService.DOCUMENT_PAGE_SIZE.getKey(),
+                ByteSizeValue.ofBytes(randomIntBetween(1000, randomFrom(10000, 100000, 1000000)))
```

Comment:
I don't think we need randomness here, let's just use the default (i.e. `PersistedClusterStateService.DOCUMENT_PAGE_SIZE.get(Settings.EMPTY)`).